docs/tutorial/progress-bar.md

```diff
@@ -31,30 +31,44 @@ currently at 63% towards completion, you would call it as
 `setProgressBar(0.63)`.
 
 Setting the parameter to negative values (e.g. `-1`) will remove the progress
-bar, whereas setting it to values greater than `1` (e.g. `2`) will switch the
-progress bar to indeterminate mode (Windows-only -- it will clamp to 100%
-otherwise). In this mode, a progress bar remains active but does not show an
-actual percentage. Use this mode for situations when you do not know how long
-an operation will take to complete.
+bar. Setting it to a value greater than `1` will indicate an indeterminate progress bar
+in Windows or clamp to 100% in other operating systems. An indeterminate progress bar
+remains active but does not show an actual percentage, and is used for situations when
+you do not know how long an operation will take to complete.
 
 See the [API documentation for more options and modes][setprogressbar].
 
 ## Example
 
-Starting with a working application from the
-[Quick Start Guide](quick-start.md), add the following lines to the
+Add the following lines to the
 `main.js` file:
 
 ```javascript fiddle='docs/fiddles/features/progress-bar'
 const { BrowserWindow } = require('electron')
 const win = new BrowserWindow()
 
-win.setProgressBar(0.5)
+const INCREMENT = 0.02
+const INTERVAL_DELAY = 100 // ms
+let c = 0
+
+// create an interval timer that fires every INTERVAL_DELAY
+// see docs at https://nodejs.org/api/timers.html#timers_setinterval_callback_delay_args
+const interval = setInterval(() => {
+    // update progress bar to next value
+    win.setProgressBar(c)
+
+    if (c > 1) {
+        // progress bar has reached full so reset it and stop the timer
+        win.setProgressBar(-1)
```

Comment:
It might be nice to pause for a few seconds at 100%, and then loop? That way if the user takes a minute to find the icon it's still easy enough to see what's happening?

Reply:
I thought of that too. To show the indeterminant state. I like it and I'll do it. Thanks.